sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -48,12 +71,13 @@ bool oclHandleInvalidWorkGroupSize(const device_impl &DeviceImpl,
     // PI_INVALID_WORK_GROUP_SIZE if local_work_size is NULL and the
     // reqd_work_group_size attribute is used to declare the work-group size
     // for kernel in the program source.
-    if (!HasLocalSize && (Ver[0] == '1' || (Ver[0] == '2' && Ver[2] == '0')))
-      throw sycl::nd_range_error(
-          "OpenCL 1.x and 2.0 requires to pass local size argument even if "
-          "required work-group size was specified in the program source",
-          PI_INVALID_WORK_GROUP_SIZE);
-
+    if (Platform.get_backend() == cl::sycl::backend::opencl) {
+      if (!HasLocalSize && (Ver[0] == '1' || (Ver[0] == '2' && Ver[2] == '0')))
+        throw sycl::nd_range_error(
+            "OpenCL 1.x and 2.0 requires to pass local size argument even if "
+            "required work-group size was specified in the program source",
+            PI_INVALID_WORK_GROUP_SIZE);
+    }
     // Any OpenCL version:
```

Comment:
This error check is **not** specific to OpenCL, so remove this part of the comment.